lib/db/mem.js

```diff
@@ -1,45 +1,61 @@
-var db = []
-
-function match(event, filter) {
-  var filterNames = Object.keys(filter)
-  for (var i = 0; i < filterNames.length; i++) {
-    var name = filterNames[i]
-    if (event[name] && event[name] === filter[name]) {
-      continue
-    }
-    else {
-      return false
+var jws = require('jws')
+
+function match(filter, str) {
+  try { // too lazy
+    var event = JSON.parse(jws.decode(str).payload)
+    var filterNames = Object.keys(filter)
+    for (var i = 0; i < filterNames.length; i++) {
+      var name = filterNames[i]
+      if (event[name] && event[name] === filter[name]) {
+        continue
+      }
+      else {
+        return false
+      }
     }
+    return true
+  } catch (e) {
+    return false
```

Comment:
It would be nicer that way, but `some` will return true if *any* of the keys match. We want true only if *all* of them to match. We could use a `reduce` though.